packages/firestore/test/unit/specs/spec_test_components.ts

```diff
@@ -66,12 +67,17 @@ export class MockMemoryPersistence extends MemoryPersistence {
     ) => PersistencePromise<T>
   ): Promise<T> {
     if (this.injectFailures) {
-      return Promise.reject(
-        new IndexedDbTransactionError(new Error('Simulated retryable error'))
-      );
-    } else {
-      return super.runTransaction(action, mode, transactionOperation);
+      if (this.injectFailures[action as PersistenceAction] === undefined) {
+        throw fail('Failure mode not specified for action: ' + action);
+      }
+      if (this.injectFailures[action as PersistenceAction]) {
+        return Promise.reject(
+          new IndexedDbTransactionError(new Error('Simulated retryable error'))
```

Comment:
Consider including the value of `action` in the error message. This may assist with debugging to figure out which action caused the exception to be thrown.

Reply:
Good idea. Future me will thank you as I am debugging test assertions.